src/cargo/core/compiler/build_plan.rs

```diff
@@ -0,0 +1,136 @@
+//! A graph-like structure used to represent the rustc commands to build the project and the
+//! interdependencies between them.
+//!
+//! The BuildPlan structure is used to store the dependency graph of a dry run so that it can be
+//! shared with an external build system. Each Module in the BuildPlan comprises a single
+//! subprocess and defines the build environment, the outputs produced by the subprocess, and the
+//! dependencies on other Modules.
+
+use std::collections::BTreeMap;
+
+use super::{Context, Unit};
+use super::context::OutputFile;
+use util::{internal, CargoResult, ProcessBuilder};
+use std::sync::Arc;
+use std::path::PathBuf;
+use serde_json;
+
+#[derive(Debug, Serialize)]
+struct Module {
+    display_name: String,
```

Comment:
Instead of `display_name`, I suggest including 

```
    pub package_id: PackageId,
    pub target: Target,
```

Reply:
It's true they won't work with `Default`, yeah, but could `Default` be removed? Using `PackageId`, `Target`, and `Profile` more closely match what Cargo is internally doing and using to represent data structures which is probably the best way to go for now to ensure it's all lossless